src/pages/settings/Wallet/ActivateCardPage.js

```diff
@@ -0,0 +1,175 @@
+        }
+
+        if (!cardList.isLoading && lodashGet(cardList, 'physical.state', 0) === CARD_ACTIVATED_STATE) {
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
+        }
+    }, [cardList, domain]);
+
+    /**
+     * Set form error if there is one for cardID in onyx
+     */
+    useEffect(() => {
+        if (isUndefined(lodashGet(cardList[cardID], 'errors', undefined))) {
+            setFormError('');
+        } else {
+            setFormError(ErrorUtils.getLatestErrorMessage(cardList[cardID]));
+        }
+    }, [cardList, cardID]);
+
+    /**
+     * Update lastPressedDigit with value that was pressed on BigNumberPad.
+     *
+     * NOTE: If the same digit is pressed twice in a row, append it to the end of the string
+     * so that useEffect inside MagicCodeInput will be triggered by artificial change of the value.
+     *
+     * @param {String} key
+     */
+    const updateLastPressedDigit = useCallback(
+        (key) => {
+            if (lastPressedDigit === key) {
+                setLastPressedDigit(lastPressedDigit + key);
+            } else {
+                setLastPressedDigit(key);
+            }
+        },
+        [lastPressedDigit],
+    );
+
+    /**
+     * Handle card activation code input
+     *
+     * @param {String} text
+     */
+    const onCodeInput = (text) => {
+        setFormError('');
+        setActivateCardCode(text);
+    };
+
+    const submitAndNavigateToNextPage = useCallback(() => {
+        activateCardCodeInputRef.current.blur();
+        CardSettings.clearCardListErrors(cardID);
+        CardSettings.activatePhysicalExpensifyCard(Number(activateCardCode), cardID);
+    }, [activateCardCode, cardID]);
+
+    return (
+        <IllustratedHeaderPageLayout
+            title={translate('activateCardPage.activateCard')}
+            onBackButtonPress={() => Navigation.goBack()}
+            backgroundColor={themeColors.PAGE_BACKGROUND_COLORS[SCREENS.SETTINGS.PREFERENCES]}
+            illustration={LottieAnimations.Magician}
+        >
+            <Text style={[styles.mh5, styles.textHeadline]}>{translate('activateCardPage.pleaseEnterLastFour')}</Text>
+            <View style={[styles.mh5]}>
+                <MagicCodeInput
+                    disableKeyboard
+                    autoComplete="off"
+                    maxLength={ACTIVATE_CARD_CODE_DESIRED_LENGTH}
+                    name="activateCardCode"
+                    value={activateCardCode}
+                    lastPressedDigit={lastPressedDigit}
+                    onChangeText={onCodeInput}
+                    errorText={formError}
+                    ref={activateCardCodeInputRef}
+                    shouldSubmitOnComplete={false}
+                />
+            </View>
+            <View style={[styles.w100, styles.justifyContentEnd, styles.pageWrapper]}>
+                {DeviceCapabilities.canUseTouchScreen() ? <BigNumberPad numberPressed={updateLastPressedDigit} /> : null}
```

Comment:
If `canUseTouchScreen` always returns a boolean value, we can simplify a condition.
```suggestion
                {DeviceCapabilities.canUseTouchScreen() && <BigNumberPad numberPressed={updateLastPressedDigit} />}
```

Reply:
Honestly I'd default to a ternary - the underlying implementation might change and it will affect this component causing a regression.

Reply:
It is already migrated to TS and has a [strict type](https://github.com/Expensify/App/blob/main/src/libs/DeviceCapabilities/canUseTouchScreen/types.ts). So, we can expect that it returns boolean forever. However, it is not something critical. Up to you.